cpp/cmake/install/FindThrust.cmake

```diff
@@ -0,0 +1,64 @@
+# FindThrust
+# ---------
+#
+# Try to find Thrust
+#
+# Uses Thrust_ROOT in the cache variables or in the environment as a hint where to search
+#
+# IMPORTED Targets
+# ^^^^^^^^^^^^^^^^
+#
+# This module defines a basic `thrust_create_target` function as provided otherwise by the newer
+# Thrust >= 1.9.10 included configs.
+#
+# Result Variables
+# ^^^^^^^^^^^^^^^^
+#
+# This module defines the following variables:
+# ~~~
+# ``Thrust_FOUND`` system has Thrust
+# ``Thrust_INCLUDE_DIRS`` the Thrust include directories
+# ~~~
+
+include(FindPackageHandleStandardArgs)
+
+# try to find Thrust via installed config first
+find_package(Thrust QUIET CONFIG)
```

Comment:
Can't we use CPM for this?

Reply:
We do for building, but this file is generated for the install scripts similar to what [RMM does](https://github.com/rapidsai/rmm/blob/94b083a61bec83167e2132917a53ef530a0d63ce/CMakeLists.txt#L115-L116).

That said, I'm not 100% certain on how CMake-generated installation scripts should work in practice, so this could be wrong.

Reply:
Got it, I'm less familiar with how the install side of the cmake script is supposed to work.